extensions/interactions/ItemSelectionInput/validator.js

```diff
@@ -89,15 +89,60 @@ oppia.filter('oppiaInteractiveItemSelectionInputValidator', [
         baseInteractionValidationService.getAnswerGroupWarnings(
           answerGroups, stateName));
 
-      if (!defaultOutcome ||
-          $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+      var selectedChoices = [];
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              var choicePreviouslySelected = (
+                selectedChoices.indexOf(ruleSpecs[j].inputs.x[k]) !== -1);
+              if (!choicePreviouslySelected) {
+                selectedChoices.push(ruleSpecs[j].inputs.x[k]);
+              }
+            }
+          }
+        }
+      }
+      if (selectedChoices.length < numChoices) {
+        if (!defaultOutcome ||
+            $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+          warningsList.push({
+            type: WARNING_TYPES.ERROR,
+            message: (
+              'Please clarify the default outcome so it is less confusing to ' +
+              'the user.')
+          });
+        }
+      }
+      // For situations where max choices is 1,
+      // ensure that number of answers matches number of choices
+      if (maxAllowedCount === 1 && answerGroups.length > numChoices) {
```

Comment:
But one answer group could potentially take care of more than one choice, right?